lib/sgtn-client/util/validate-util.rb

```diff
@@ -6,31 +6,31 @@ module SgtnClient
   class ValidateUtil
 
       def self.validate_config()
-        env = Config.default_environment
-        SgtnClient.logger.debug "[ValidateUtil][validate_config] env = #{env}"
-        messages = "\n"
+        SgtnClient.logger.debug "[ValidateUtil][validate_config]"
+        messages = ""
 
-        mode = Config.configurations[env]["mode"]
+        mode = SgtnClient.config.mode
         if mode != 'sandbox' && mode != 'live'
           messages =  messages + "Configuration[mode] has to be 'sandbox' or 'live'!\n"
         end
 
-        #version = Config.configurations[env]["version"]
-        #if version.is_a? Integer
-          #messages = messages +  "Configuration[version] has to be standard as '#.#.#, e.g '1.0.0'!\n"
-        #end
+        version = SgtnClient.config.version
+        SgtnClient.config.version = version.to_s
+        if version.to_s !~ /\A(\d+\.)*\d+\z/
+          messages = messages +  "Configuration[version] has to be standard as '#.#.#, e.g '1.0.0'!\n"
+        end
         
-        cache_expiry_period = Config.configurations[env]["cache_expiry_period"]
+        cache_expiry_period = SgtnClient.config.cache_expiry_period
         if cache_expiry_period != nil && (cache_expiry_period.is_a? Integer) == false
           messages = messages +  "Configuration[cache_expiry_period] has to be a number!\n"
         end
         
-        disable_cache = Config.configurations[env]["disable_cache"]
+        disable_cache = SgtnClient.config.disable_cache
         if disable_cache != nil && disable_cache != false && disable_cache != true
           messages = messages +  "Configuration[disable_cache] has to be a 'true' or 'false'!\n"
         end
         
-        if messages != "\n"
+        if messages != ""
```

Comment:
Could you change "messages" to something more descriptive like "error_messages"? Also, could you make this a list/array instead of one string and pass to MissingConfig?

This can be in a separate PR if you prefer.

Reply:
Yes. Let's change it in another PR.